docs/design/assetgeneration.md

```diff
@@ -63,6 +39,22 @@ type Asset interface{
 }
 ```
 
+## Writable Asset
```

Comment:
Is there an example of non-writable assets?

Reply:
The dependencies of InstallConfig are all non-writable assets--baseDomain, clusterName, platform, etc.

Reply:
What is stopping them from being writable?

Reply:
There is nothing stopping them from being writeable other than that there is no need for them to be writeable.

Let's take the baseDomain as an example. If it were writeable, then we would need to either (1) name a file for the base domain and a format for the data in the file or (2) have the `Files` function return an empty slice. In the first case, we would being doing it for a file that is never consumed by anything. In the second case, we would be adding an unnecessary function to the asset.

The `WriteableAsset` interface is nothing more than a convenience to keep unnecessary functions out of assets that do not generate any files.